android/app/src/main/java/com/coboltforge/dontmind/multivnc/EditBookmarkActivity.java

```diff
@@ -147,54 +141,47 @@ private void updateViewsFromBookmark() {
 				break;
 			}
 
-		if(bookmark.getUseRepeater())
-			repeaterText.setText(bookmark.getRepeaterId());
+		if(bookmark.useRepeater)
+			repeaterText.setText(bookmark.repeaterId);
 	}
 	
 	
 	private void updateBookmarkFromViews() {
-	
-		bookmark.setAddress(ipText.getText().toString());
+
+		bookmark.address = ipText.getText().toString();
 		try
 		{
-			bookmark.setPort(Integer.parseInt(portText.getText().toString()));
+			bookmark.port = Integer.parseInt(portText.getText().toString());
 		}
 		catch (NumberFormatException nfe)
 		{
 			
 		}
-		bookmark.setNickname(bookmarkNameText.getText().toString());
-		bookmark.setUserName(usernameText.getText().toString());
-		bookmark.setPassword(passwordText.getText().toString());
-		bookmark.setKeepPassword(checkboxKeepPassword.isChecked());
-		bookmark.setUseLocalCursor(true); // always enable
-		bookmark.setColorModel(((COLORMODEL)colorSpinner.getSelectedItem()).nameString());
+		bookmark.nickname = bookmarkNameText.getText().toString();
+		bookmark.userName = usernameText.getText().toString();
+		bookmark.password = passwordText.getText().toString();
+		bookmark.keepPassword = checkboxKeepPassword.isChecked();
+		bookmark.useLocalCursor = true; // always enable
+		bookmark.colorModel = ((COLORMODEL)colorSpinner.getSelectedItem()).nameString();
 		if (repeaterText.getText().length() > 0)
 		{
-			bookmark.setRepeaterId(repeaterText.getText().toString());
-			bookmark.setUseRepeater(true);
+			bookmark.repeaterId = repeaterText.getText().toString();
+			bookmark.useRepeater = true;
 		}
 		else
-		{			
-			bookmark.setUseRepeater(false);
+		{
+			bookmark.useRepeater = false;
 		}
 	}
 	
 	
 	
 	private void saveBookmark(ConnectionBean conn) 	{
-		SQLiteDatabase db = database.getWritableDatabase();
-		db.beginTransaction();
 		try {
-			Log.d(TAG, "Saving bookmark for conn " + conn.get_Id());
-			conn.save(db);
-			db.setTransactionSuccessful();
+			database.getConnectionDao().save(conn);
```

Comment:
Removed the log accidentally. Reverted it.


For try-block, I don't think we will get an exception here because we don't have any constraints in our database and anything more fatal like corrupt database would have been noticed before reaching here. But I am not 100% sure.

Reply:
Won't hurt either.